sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/util/Context.java

```diff
@@ -212,63 +205,11 @@ public Optional<Object> getData(Object key) {
     }
 
     /**
-     * Gets the {@link Context Contexts} in the chain of Contexts that this Context is the tail.
-     *
-     * @return The Contexts, in oldest-to-newest order, in the chain of Contexts that this Context is the tail.
-     */
-    Context[] getContextChain() {
-        Context[] chain = new Context[contextCount];
-
-        int chainPosition = contextCount - 1;
-
-        for (Context pointer = this; pointer != null; pointer = pointer.parent) {
-            chain[chainPosition--] = pointer;
-
-            // If the contextCount is 1 that means the next parent Context is the NONE Context.
-            // Break out of the loop to prevent a meaningless check.
-            if (pointer.contextCount == 1) {
-                break;
-            }
-        }
-
-        return chain;
-    }
-
-    /**
-     * Merges two {@link Context Contexts} into a new {@link Context}.
-     *
-     * @param into Context being merged into.
-     * @param from Context being merged.
-     *
-     * @return A new Context that is the merged Contexts.
+     * Get the total size of this {@link Context}.
      *
-     * @throws NullPointerException If either {@code into} or {@code from} is {@code null}.
+     * @return The total size of this {@link Context}.
      */
-    public static Context mergeContexts(Context into, Context from) {
-        Objects.requireNonNull(into, "'into' cannot be null.");
-        Objects.requireNonNull(from, "'from' cannot be null.");
-
-        // If the 'into' Context is the NONE Context just return the 'from' Context.
-        // This is safe as Context is immutable and prevents needing to create any new Contexts and temporary arrays.
-        if (into == Context.NONE) {
-            return from;
-        }
-
-        // Same goes the other way, where if the 'from' Context is the NONE Context just return the 'into' Context.
-        if (from == Context.NONE) {
-            return into;
-        }
-
-        Context[] contextChain = from.getContextChain();
-
-        Context returnContext = into;
-
-        for (Context toAdd : contextChain) {
-            if (toAdd != null) {
-                returnContext = returnContext.addData(toAdd.getKey(), toAdd.getValue());
-            }
-        }
-
-        return returnContext;
+    public int getContextCount() {
```

Comment:
I would suggest we call this something like `getSize()` or `getLength()`. Let's discuss in teams chat.

Reply:
We can go with `getSize()` or just `size()` (to match the method name on `List` interface). Currently, it returns the total number of items in the linked list but should it return the total number of unique items in the list? For e.g. `new Context("a", "a-value").addData("a", "b-value")` has a size of 2 but actually, there's no way to get `a-value` with key "a" and the practical size is just 1.